File: lib/models/match.dart
class Match {
  final int? id;
  final int championshipId;
  final String player1;
  final String player2;
  final String game;
  final MatchStatus status;
  final String? winner;
  final int player1Score;
  final int player2Score;
  final DateTime? startedAt;
  final DateTime? finishedAt;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Match({
    this.id,
    required this.championshipId,
    required this.player1,
    required this.player2,
    required this.game,
    this.status = MatchStatus.pending,
    this.winner,
    this.player1Score = 0,
    this.player2Score = 0,
    this.startedAt,
    this.finishedAt,
    this.createdAt,
    this.updatedAt,
  });

  factory Match.fromJson(Map<String, dynamic> json) {
    MatchStatus parseStatus(String? status) {
      switch (status) {
        case 'started':
          return MatchStatus.started;
        case 'finished':
          return MatchStatus.finished;
        default:
          return MatchStatus.pending;
      }
    }

    return Match(
      id: json['id'],
      championshipId: json['championship_id'] ?? 0,
      player1: json['player1'],
      player2: json['player2'],
      game: json['game'],
      status: parseStatus(json['status']),
      winner: json['winner'],
      player1Score: json['player1_score'] ?? 0,
      player2Score: json['player2_score'] ?? 0,
      startedAt: json['started_at'] != null
          ? DateTime.parse(json['started_at'])
          : null,
      finishedAt: json['finished_at'] != null
          ? DateTime.parse(json['finished_at'])
          : null,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    String statusString() {
      switch (status) {
        case MatchStatus.started:
          return 'started';
        case MatchStatus.finished:
          return 'finished';
        default:
          return 'pending';
      }
    }

    return {
      if (id != null) 'id': id,
      'championship_id': championshipId,
      'player1': player1,
      'player2': player2,
      'game': game,
      'status': statusString(),
      if (winner != null) 'winner': winner,
      'player1_score': player1Score,
      'player2_score': player2Score,
    };
  }
}

enum MatchStatus {
  pending,
  started,
  finished,
}

File: backend/internal/models/match.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type MatchStatus string

const (
	MatchStatusPending  MatchStatus = "pending"
	MatchStatusStarted  MatchStatus = "started"
	MatchStatusFinished MatchStatus = "finished"
)

type Match struct {
	ID            uint           `json:"id" gorm:"primaryKey"`
	ChampionshipID uint          `json:"championship_id" gorm:"not null;index"`
	Player1       string         `json:"player1" gorm:"not null"`
	Player2       string         `json:"player2" gorm:"not null"`
	Game          string         `json:"game" gorm:"not null"`
	Status        MatchStatus    `json:"status" gorm:"type:varchar(20);default:'pending';not null"`
	Winner        *string        `json:"winner" gorm:"default:null"` // Nullable, wird erst beim Beenden gesetzt
	Player1Score  int            `json:"player1_score" gorm:"default:0;not null"`
	Player2Score  int            `json:"player2_score" gorm:"default:0;not null"`
	StartedAt     *time.Time     `json:"started_at" gorm:"default:null"`
	FinishedAt    *time.Time     `json:"finished_at" gorm:"default:null"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `json:"-" gorm:"index"`
	
	// Relations
	Championship Championship `json:"championship,omitempty" gorm:"foreignKey:ChampionshipID"`
}

